### PR TITLE
Alternative PR for: Some bytes/str issues in py3 w/ zlib and json

### PR DESCRIPTION
**Pull request by rumbin**
This is in essence the same as #2034, however, it is rebased now and reduced to what is really needed.

### CODE REVIEW

File: tests/utils_tests.py [hunk 1]
```diff
@@ -45,3 +45,10 @@ def test_base_json_conv(self):
     def test_parse_human_timedelta(self, mock_now):
         mock_now.return_value = datetime(2016, 12, 1)
         self.assertEquals(parse_human_timedelta('now'), timedelta(0))
+
+    def test_zlib_compression(self):
+        json_str = """{"test": 1}"""
+        blob = zlib_compress(json_str)
+        got_str = zlib_uncompress_to_string(blob)
```
Comment (mistercrunch) [inline tests/utils_tests.py]: `zlib_decompress_to_string` this is what breaks the build

File: superset/utils.py [hunk 2]
```diff
@@ -572,3 +574,46 @@ def setup_cache(app, cache_config):
     """Setup the flask-cache on a flask app"""
     if cache_config and cache_config.get('CACHE_TYPE') != 'null':
         return Cache(app, config=cache_config)
+
+
+
+def zlib_compress(data):
```
Comment (mistercrunch) [inline superset/utils.py]: too many blank lines, should be 2 blank lines above a function (PEP8)

File: superset/utils.py [hunk 3]
```diff
@@ -572,3 +574,46 @@ def setup_cache(app, cache_config):
...
+            decompressed = zlib.decompress(blob)
+        else:
+            decompressed = zlib.decompress(bytes(blob, "utf-8"))
+
+        if isinstance(decompressed, str):
+            return decompressed
+        return decompressed.decode("utf-8")
+    else:
+        return zlib.decompress(blob)
+
```
Comment (mistercrunch) [inline superset/utils.py]: remove blank line

**Review by mistercrunch — CHANGES_REQUESTED** [review]

File: superset/utils.py [hunk 4]
```diff
@@ -572,3 +574,46 @@ def setup_cache(app, cache_config):
...
+    uncompress things to a string in a py2/3 safe fashion
+    >>> json_str = '{"test": 1}'
+    >>> blob = zlib_compress(json_str)
+    >>> got_str = zlib_uncompress_to_string(blob)
+    >>> got_str == json_str
+    True
+    """
+
+    if PY3K:
+        decompressed = ""
```
Comment (xrmx) [inline superset/utils.py]: You don't need this

File: superset/utils.py [hunk 5]
```diff
@@ -572,3 +574,46 @@ def setup_cache(app, cache_config):
...
+        if isinstance(blob, bytes):
+            decompressed = zlib.decompress(blob)
+        else:
+            decompressed = zlib.decompress(bytes(blob, "utf-8"))
+
+        if isinstance(decompressed, str):
```
Comment (xrmx) [inline superset/utils.py]: python3 documentation says that zlib.decompress returns bytes

File: superset/utils.py [hunk 6]
```diff
@@ -572,3 +574,46 @@ def setup_cache(app, cache_config):
...
+    compress things in a py2/3 safe fashion
+
+    >>> json_str = '{"test": 1}'
+    >>> blob = zlib_compress(json_str)
+    """
+
+    if PY3K:
+        if isinstance(data, str):
+            return zlib.compress(bytes(data, "utf-8"))
+        else:
```
Comment (xrmx) [inline superset/utils.py]: We may put a return zlib.compress(data) outside the if PY3K instead of the following 4 lines.

File: superset/utils.py [hunk 7]
```diff
@@ -572,3 +574,35 @@ def setup_cache(app, cache_config):
...
+def zlib_compress(data):
+    """
+    Compress things in a py2/3 safe fashion
+    >>> json_str = '{"test": 1}'
+    >>> blob = zlib_compress(json_str)
+    """
+    if PY3K:
+        if isinstance(data, str):
+            return zlib.compress(bytes(data, "utf-8"))
+        else:
```
Comment (xrmx) [inline superset/utils.py]: You can drop the else, it'll we handled by the last return

**Comment by mistercrunch** [comment]
Thanks for getting this through!

**Pull request merged**